tests/subsys/fs/common/test_fs_open_flags.c

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @filesystem
+ * @brief test_filesystem
+ * Tests the fs_open flags
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <fs/fs.h>
+#include <string.h>
+
+/* Path for testr file should be provided by test runner and should start
+ * with mount point.
+ */
+extern char *test_fs_open_flags_file_path;
+
+static const char something[] = "Something";
+static char buffer[sizeof(something)];
+#define RDWR_SIZE sizeof(something)
+
+struct test_state {
+	/* Path to file */
+	char *file_path;
+	struct fs_file_t file;
+	/* Read write buffer info */
+	const char *write;
+	int write_size;
+	char *read;
+	int read_size;
+};
+
+/* ZEQ decides whether test completed successfully and prints appropriate
+ * information.
+ */
+static void ZEQ(int ret, int expected)
```

Comment:
Use lower case for all these function names: all-caps belongs to macros only.

The trick of `#DEFINE FOO(a) FOO(a,x)` is confusing because the "same" function appears to have two signatures; using the correct style for these names will eliminate that confusion.